e2e/content-services/pages/permissions.page.ts

```diff
@@ -15,29 +15,89 @@
     searchUserInput = element(by.id('searchInput'));
     searchResults = element(by.css('#adf-add-permission-authority-results #adf-search-results-content'));
-    addButton = element(by.id('add-permission-dialog-confirm-button'));
+    addButton = element(by.css('[data-automation-id="add-permission-dialog-confirm-button"]'));
     permissionInheritedButton = element.all(by.css('.app-inherit_permission_button button')).first();
     noPermissions = element(by.id('adf-no-permissions-template'));
     deletePermissionButton = element(by.css(`button[data-automation-id='adf-delete-permission-button']`));
     permissionDisplayContainer = element(by.id('adf-permission-display-container'));
     closeButton = TestElement.byCss('#add-permission-dialog-close-button');
 
     async changePermission(name: string, role: string): Promise<void> {
-        await this.addPermissionsDialog.clickRoleDropdownByUserOrGroupName(name);
-        await this.addPermissionsDialog.selectOption(role);
         await browser.sleep(500);
-        await this.dataTableComponentPage.checkRowIsNotSelected('Authority ID', name);
+        await this.clickRoleDropdownByUserOrGroupName(name);
+        await new DropdownPage().selectOption(role);
+        await this.dataTableComponentPage.checkRowByContentIsNotSelected(name);
+    }
+
+    async checkUserIsAdded(id: string) {
+        const userOrGroupName = TestElement.byCss('div[data-automation-id="' + id + '"]');
+        await userOrGroupName.waitPresent();
+    }
+
+    async getRoleCellValue(username: string): Promise<string> {
+        const locator = this.dataTableComponentPage.getCellByRowContentAndColumn('Users and Groups', username, 'Role');
+        return BrowserActions.getText(locator);
+    }
+
+    async clickRoleDropdownByUserOrGroupName(name: string): Promise<void> {
+        const row = this.dataTableComponentPage.getRow('Users and Groups', name);
+        await row.click();
+        await BrowserActions.click(row.element(by.css('[id="adf-select-role-permission"] .mat-select-trigger')));
+    }
+
+    async clickDeletePermissionButton(username: string): Promise<void> {
+        const userOrGroupName = TestElement.byCss(`[data-automation-id="adf-delete-permission-button-${username}"]`);
+        await userOrGroupName.waitPresent();
+        await userOrGroupName.click();
+    }
+
+    async checkUserIsDeleted(username: string): Promise<void> {
+        const userOrGroupName = TestElement.byCss('div[data-automation-id="' + username + '"]');
+        await userOrGroupName.waitNotPresent();
+    }
+
+    async noPermissionContent(): Promise<string> {
+        const noPermission = TestElement.byCss('.adf-no-permission__template--text');
+        return noPermission.getText();
+    }
+
+    async waitVisible(): Promise<void> {
+        await TestElement.byId(this.rootElement).waitVisible();
+    }
+
+    async waitTillContentLoads(): Promise<void> {
```

Comment:
what's the purpose of isNotDisplayed? you're only using it for the log message? 
on the catch block, if isNotDisplayed is false, does this mean **IT IS** displayed? (I don't know who's "it" in this :) )

can you please explain what this method does?